File: backend/src/main/java/be/technobel/parsemaster/validation/validator/NotBlankValidator.java
package be.technobel.parsemaster.validation.validator;

import be.technobel.parsemaster.dto.ErrorFieldDTO;
import be.technobel.parsemaster.exception.InvalidParamException;
import be.technobel.parsemaster.validation.constraint.NotBlank;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class NotBlankValidator implements ConstraintValidator<NotBlank, String> {
  private String field;
  private String message;

  @Override
  public void initialize(NotBlank annotation) {
    field = annotation.field();
    message = annotation.message();
  }

  @Override
  public boolean isValid(String value, ConstraintValidatorContext context) {
    if(value == null || !value.isBlank()) return true;
    throw new InvalidParamException(new ErrorFieldDTO(field, message));
  }
}

File: src/main/kotlin/be/technobel/playzone/pl/validation/validators/NotBlankValidator.kt
package be.technobel.playzone.pl.validation.validators

import be.technobel.playzone.pl.rest.FormError
import be.technobel.playzone.pl.rest.InvalidParamException
import be.technobel.playzone.pl.validation.constraints.NotBlank
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class NotBlankValidator : ConstraintValidator<NotBlank, String> {
	private lateinit var field: String
	private lateinit var message: String

	override fun initialize(constraintAnnotation: NotBlank) {
		field = constraintAnnotation.field
		message = constraintAnnotation.message
	}

	override fun isValid(value: String, context: ConstraintValidatorContext): Boolean {
		return if (value.isNotBlank()) true else throw InvalidParamException(
			FormError(field, "$field $message")
		)
	}
}